File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.2/Java/src/Main.java
public class Main {
  public static int count(int[] list, int start) {
    if (start >= list.length) {
      return 0;
    }

    return 1 + count(list, start + 1);
  }

  public static void main(String[] args) {
    System.out.println(count(new int[]{1, 3, 5, 7}, 0));
    System.out.println(count(new int[]{2, 4, 6, 10, 12}, 0));
    System.out.println(count(new int[]{3, 5, 7, 9, 13, 15}, 0));
  }
}

File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.2/Rust/src/main.rs
fn count(list: &[i32]) -> usize {
    if list.is_empty() {
        0
    } else {
        1 + count(&list[1..])
    }
}

fn main() {
    println!("{}", count(&[1, 3, 5, 7]));
    println!("{}", count(&[2, 4, 6, 10, 12]));
    println!("{}", count(&[3, 5, 7, 9, 13, 15]));
}
